docs/posts/dbt-ibis/index.qmd

```diff
@@ -0,0 +1,93 @@
+
+
+# Why dbt and Ibis go great together
+[dbt-ibis](https://github.com/binste/dbt-ibis) offers a more lightweight and compatible alternative, which allows you to write dbt models using Ibis. dbt-ibis transparently converts your Ibis statements into SQL and then hands it over to dbt. Your database does not need to have Python support for this as everything is executed in the same process as dbt. Rewriting the above SQL model in Ibis we get:
+
+```python
+from dbt_ibis import depends_on, ref
+
+
+@depends_on(ref("stg_customers"), ref("stg_orders"))
+def model(customers, orders):
+    customer_orders = orders.group_by("customer_id").aggregate(
+        first_order=orders["order_date"].min(),
+        most_recent_order=orders["order_date"].max(),
+        number_of_orders=orders.count(),
+    )
+    # Add first_name and last_name
+    customer_orders = customers.join(customer_orders, "customer_id", how="left")
+    return customer_orders.select(
+            "customer_id",
+            "first_name",
+            "last_name",
+            "first_order",
+            "most_recent_order",
+            "number_of_orders",
+        )
+```
+
+Using Ibis instead of SQL for dbt models brings you many advantages:
+
+* Type checks and validation before your code is even executed in a database
+* More composable as you can break down complex queries into smaller pieces
+* Better reusability of code. Although dbt allows you to use [Jinja and macros](https://docs.getdbt.com/docs/build/jinja-macros), which is an improvement over plain SQL, this gets you only so far. String manipulation is inherently fragile. With dbt-ibis, you can easily share common code between models.
+* Your dbt models become backend agnostic which reduces lock-in to a specific database. Furthermore, you get the possibility of building a [multi-engine data stack](https://juhache.substack.com/p/n-engines-1-language?utm_source=post-email-title&publication_id=1211981&post_id=137718100). For example, you could use DuckDB for small to medium workloads and Snowflake for heavy workloads and as an end-user and BI layer leveraging its governance features. Depending on the size of your warehouse, this can result in significant cost savings.
+* Unit test your code with your favorite Python testing frameworks such as pytest.
+
+In addition, you can stick to the tool (Ibis) you like, no matter if you're writing an ingestion pipeline, a dbt model to transform the data in your data warehouse, or conduct an ad-hoc analysis in a Jupyter notebook.
+
+Be aware that a current limitation of dbt-ibis is, that you cannot connect to the database from within your dbt models, i.e. you purely use Ibis to construct a SELECT statement. You cannot execute statements and act based on the results.
+
+# Further readings
```

Comment:
might want a further call to action either here or below -- is there a good link to point them to for getting started w/ dbt-ibis and trying it out?